docs/spec_draft.md

```diff
@@ -1808,6 +1809,64 @@ where `i[d] = j[permutation[d]]`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.while
+
+### Semantics
+Produces the output from executing `body` function 0 or more times till the
+`cond` function outputs `true`. More formally, `body` function is executed once
+every time the `cond` function returns `true` and all execution stops once
+`cond` function returns `false`. For the first iteration both `cond` and `body`
+arguments are initialzed with the vaules of arguments of the `while` operation,
+for subsequent iterations they are fed with the output of `body` from the
+previous itertaion.
+
+### Inputs
+
+| Name       | Type                                             |
+|------------|--------------------------------------------------|
+| `operands` | variadic number of tensors of any supported type |
+| `cond`     | `function`                                       |
+| `body`     | `function`                                       |
+
+### Outputs
+
+| Name      | Type                                             |
+|-----------|--------------------------------------------------|
+| `results` | variadic number of tensors of any supported type |
+
+### Constraints
+
+  * (C1) Size of `operands` is at least 1.
```

Comment:
It seems that HLO does accept 0 size of operands. Shouldn't we align with that?

Reply:
Nice catch! Similarly, `if` and `case` cannot have 0 outputs in StableHLO, but I don't think that HLO prohibits that. What should we do about that?

Reply:
> It seems that HLO does accept 0 size of operands. Shouldn't we align with that?

To me the only way to answer at such question is "what is the use case"? What are we trying to support? 
IMO our users does not include HLO : it should be about TF/JAX/PyTorch/... and what they may want to express), MHLO and HLO are rather "constraints" in that StableHLO has to ensure that before exposing anything to users it has to be able to be supported by MHLO downstream.

Reply:
I cant think of any use cases, as an empty operand list would mean, it does not return/modify anything. It can either be an empty while loop that ends immediately or an infinite loop, either of which is not useful. We can add this to the verifier and add a constraint or let this be as is. Let me know your thoughts or if there is a use case that might be possible for such a construct.

Reply:
So far, we don't have side-effecting ops in the spec, so it doesn't matter, but in the future we will, e.g. distribution ops. With distribution ops like send/recv, it's not inconceivable to have a loop body that doesn't take inputs but still does useful work. Same for if/case. Let's open a ticket to get to the bottom of this.